### PR TITLE
Get fsi compile/test standalone

### LINKED ISSUE #17962: Not all modules can be compiled/tested standalone
## Bug Description
fsi and misc cannot be compiled and/or tested standalone without failure. See https://civet.inl.gov/job/750447/.

## Steps to Reproduce
Compile fsi and misc standalone.

## Impact
Things can't compile. This is bad.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #17962

Once this is in, a general modules recipe can be added that compiles them standalone (see https://civet.inl.gov/job/750447/, which is added to this PR).

### CODE REVIEW

File: modules/modules.mk [hunk 1]
```diff
@@ -245,11 +245,20 @@ ifeq ($(FSI),yes)
   APPLICATION_DIR    := $(MOOSE_DIR)/modules/fsi
   APPLICATION_NAME   := fsi
 
-  DEPEND_MODULES     := tensor_mechanics navier_stokes
+  DEPEND_MODULES     := fluid_properties navier_stokes ray_tracing rdg tensor_mechanics
```
Comment (GiudGiud) [inline modules/modules.mk:248]: need heat conduction for navier stokes?

Comment (fdkong) [reply modules/modules.mk:248]: I am kind of puzzed. FSI requires `navier_stokes` and `tensor_mechanics`, 

and navier_stokes requires `fluid_properties` or whatever. 

Why we need to specify the requirements of `navier_stokes` in FSI? I would imagine it should be done automatically

**Review by fdkong — CHANGES_REQUESTED** [review]
It is definitely a great to have ability to build `fsi` standalone.

Any tech barrier to make the dependent chain auto transfer to next level?

For example, if B needs A, and C needs B, then we should know C also needs A by visiting this chain. 

Otherwise we have to specify that C needs B and A.

**Comment by loganharbour** [comment]
@fdkong - it is not done automatically, unfortunately. Hence why fsi will not compile standalone right now.

File: modules/modules.mk (hunk 1 above)
Comment (loganharbour) [reply modules/modules.mk:248]: @fdkong - it is not done automatically, unfortunately. Hence why fsi will not compile standalone right now.

Comment (loganharbour) [reply modules/modules.mk:248]: Well, I think it can be done somewhat automatically if the ordering at the bottom is done correctly. Let me see.

Comment (loganharbour) [reply modules/modules.mk]: Done

**Comment by loganharbour** [comment]
This is best looked at not as the diff but as the full file; I did some reordering alphabetically where possible

**Review by andrsd — DISMISSED** [review]
LGTM

**Review by fdkong — DISMISSED** [review]
Nice work

**Review by GiudGiud — DISMISSED** [review]

**Comment by loganharbour** [comment]
I ended up bailing on getting misc to test standalone - we use it to test the dynamic module loading, so we actually don't want to compile the other modules with it. This is good to go. Testing it on mastodon with FSI now.

**Review by GiudGiud — APPROVED** [review]

**Comment by loganharbour** [comment]
Just successfully compiled fsi with mastodon with these fixes.

FYI @cbolisetti: Once this gets merged and into master and mastodon is updated, I'll invalidate https://github.com/idaholab/mastodon/pull/378 and it should be good to go.

**Pull request merged**